other/text_to_speech.py

```diff
@@ -0,0 +1,137 @@
+    "John": "IVONA Eric22",
+    "Jenna": "IVONA Jennifer22",
+    "George": "IVONA Joey22",
+    "Alice": "IVONA Kimberly22",
+    "Daisy": "IVONA Salli22",
+    "Alessandra": "IVONA Carla22 (Italian)",
+    "Giovanni": "IVONA Giorgio22 (Italian)",
+    "Isabella": "IVONA Conchita22 (Spanish [Modern])",
+    "Mateo": "IVONA Enrique22 (Spanish [Modern])",
+    "Rodrigo": "IVONA Cristiano22 (Portuguese)",
+}
+
+
+def get_mp3_filename(html: str) -> Optional[str]:
+    """Returns the  mp3 file name of an HTML response from fromtexttospeech.com"""
+    if match := re.search(r"\/output\/\d+\/\d+\.mp3", html):
+        filename = match.group()
+    else:
+        filename = None
+
+    return filename
+
+
+def request_fromtexttospeech(
+    text: str = "Hello world",
+    language: str = "British",
+    voice: str = "Emma",
+) -> requests.Response:
+    """Makes POST request to fromtexttospeech and returns a requests.Response."""
+    # prepare data
+    data = {
+        "input_text": text,
+        "language": language,
+        "voice": VOICES[voice],
+        "speed": 0,
+        "action": "process_text",
+    }
+
+    headers = {
+        "Accept-Language": "en-US,en;q=0.5",
+        "Upgrade-Insecure-Requests": "1",
+        "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:66.0) Gecko/20100101 "
+        + "Firefox/66.0",
+        "Content-Type": "application/x-www-form-urlencoded",
+        "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+        "Referer": "http://www.fromtexttospeech.com/",
+        "Connection": "keep-alive",
+    }
+    # make request to www.fromtexttospeech.com
+    # it gives an html with the mp3's url
+    return requests.post("http://www.fromtexttospeech.com/", headers=headers, data=data)
+
+
+def text_to_speech(
+    text: str = "Hello world",
+    language: str = "British",
+    voice: str = "Emma",
+    audio_path: str = "output.mp3",
+) -> bool:
+    """Given a text it converts into an  MP3 audio file, and saves it in `audio_path`.
+
+    Args:
+        text (str): The text to convert
+        language (str): The voice language.
+        voice (str): The voice to use.
+        audio_path (str): Path(with file name) where to save the file.
+
+    Returns:
+        bool: True for success, False otherwise
+
+    Raises:
+        ValueError: If `language` or `voice` is not valid value.
+
+    Example:
+        text_to_speech(text="¡Hola Mundo!", language="Spanish", voice="Mateo")
+    """
+
+    if language not in LANGUAGES_VOICES:
+        raise ValueError(f"Language {language!r} not available")
+
+    if voice not in LANGUAGES_VOICES[language]:
+        raise ValueError(f"Voice {voice!r} not available")
+
+    r = request_fromtexttospeech(text, language, voice)
+    # extract the mp3's url
+    if (mp3_file := get_mp3_filename(r.text)) is None:
+        return False
+
+    audio_url = f"http://www.fromtexttospeech.com{mp3_file}"
+
+    mp3data = requests.get(audio_url)
```

Comment:
```suggestion
    mp3data = requests.get(f"http://www.fromtexttospeech.com{mp3_file}")
```

Reply:
This is not really `mp3data`.  Instead, it is a `requests.Response` so perhaps `response` would be a better variable name than `mp3data`.